lib/ThreeElementVectorLoweringPass.cpp

```diff
@@ -294,24 +275,50 @@ bool vec3ShouldBeLowered(Module &M) {
   }
 }
 
-} // namespace
+bool clspv::ThreeElementVectorLoweringPass::vec3BitcastInFunction(Function &F) {
+  for (Instruction &I : instructions(F)) {
+    if (auto *Inst = dyn_cast<CastInst>(&I)) {
+      auto *Type = Inst->getSrcTy();
+      if (Type->isPointerTy() && !Type->isOpaquePointerTy()) {
+        auto *PointeeType = Type->getNonOpaquePointerElementType();
+        if (auto *VectorType = dyn_cast<FixedVectorType>(PointeeType)) {
+          if (VectorType->getElementCount().getKnownMinValue() == 3)
```

Comment:
From my understanding this should be fixing bug with no effect. As the previous logic when it sees a vector type element, it will process the pass even when the vector type is not of size 3 which will do nothing in this case. This should fix that and only run the pass when it finds vectors of type 3.